code/game/objects/items/devices/defib.dm

```diff
@@ -574,6 +577,45 @@
 	safety = 0
 	chargetime = (1 SECONDS)
 
+/obj/item/weapon/shockpaddles/rig
+	name = "integrated resucitation pads"
+	desc = "A pair of advanced resucitation plates powered by a rig's internal power cell, able to penetrate thick clothing."
+	chargecost = 50
+	combat = 1
+	icon_state = "defibpaddles0"
+	item_state = "defibpaddles0"
+	cooldowntime = (3 SECONDS)
+
+/obj/item/weapon/shockpaddles/rig/get_cell()
+	if(istype(loc, /obj/item/rig_module))
+		var/obj/item/rig_module/RM = src.loc
+		if(istype(RM))
+			return RM.get_cell()
+
+	return FALSE
+
+/obj/item/weapon/shockpaddles/rig/check_charge(var/charge_amt)
+	var/obj/item/weapon/cell/C = get_cell()
+
+	return (C?.check_charge(charge_amt))
+
+/obj/item/weapon/shockpaddles/rig/checked_use(var/charge_amt)
+	var/obj/item/weapon/cell/C = get_cell()
+
+	return (C?.checked_use(charge_amt))
+
+/obj/item/weapon/shockpaddles/rig/universal/can_defib(mob/living/carbon/human/H)
+	if((H.species.flags & NO_DEFIB))
+		return "buzzes, \"Incompatible physiology. Operation aborted.\""
+
+	if(H.stat != DEAD)
+		return "buzzes, \"Patient is not in a valid state. Operation aborted.\""
+
+	if(!check_contact(H))
+		return "buzzes, \"Patient's chest is obstructed. Operation aborted.\""
+
+	return null
+
```

Comment:
```suggestion

/obj/item/weapon/shockpaddles/rig/universal/can_use(mob/user, mob/M)
	if(busy)
		return FALSE
	if(!check_charge(chargecost))
		to_chat(user, "<span class='warning'>\The [src] doesn't have enough charge left to do that.</span>")
		return FALSE
	if(cooldown)
		to_chat(user, "<span class='warning'>\The [src] are re-energizing!</span>")
		return FALSE
	return TRUE
```
Not actually sure how user might be populated here, so the to_chat()s may end up being meaningless

Reply:
If the user being an... AI? pAI? would actually pass the parent type, then feel free to ignore this